src/main/java/org/znerd/confluence/client/http/ConfluenceRestClient.java

```diff
@@ -426,6 +444,31 @@ public void deleteLabelFromPage(String contentId, String labelName) {
         sendRequest(deleteLabelFromPageRequest, (ignored) -> null);
     }
     
+    public ArrayList<ConfluenceLabel> getLabelsFromPage(String contentId) {
+
+        ArrayList<ConfluenceLabel> labels = new ArrayList<ConfluenceLabel>();   
+        HttpGet getLabelsByContentIdRequest = this.httpRequestFactory.getLabelsByContentIdRequest(contentId);
+
+        sendRequestAndFailIfNot20x(getLabelsByContentIdRequest, (response) -> {
+            JsonNode jsonNode = parseJsonResponse(response);
+    
+            int numberOfResults = jsonNode.get("size").asInt();
+            if (numberOfResults == 0) {
+                throw new NotFoundException();
```

Comment:
Is my understanding correct that when a page has no labels, then a `NotFoundException` is thrown?

In my view, a page does not need to have labels, so I suggest to change the approach, and rather return an empty list, e.g. by returning `Collections.emptyList()`.